source/extensions/filters/http/istio_stats/istio_stats.cc

```diff
@@ -761,6 +762,34 @@ class IstioStatsFilter : public Http::PassThroughFilter,
         service_host_name = service_host;
       }
     }
+
+    // Implements fallback from using the namespace from SAN if available to
+    // using peer metadata, otherwise.
+    absl::string_view peer_san;
+    const Ssl::ConnectionInfoConstSharedPtr ssl_info =
+        config_->reporter() == Reporter::ServerSidecar
+            ? info.downstreamAddressProvider().sslConnection()
+            : (info.upstreamInfo()
+                   ? info.upstreamInfo()->upstreamSslConnection()
+                   : nullptr);
+    if (ssl_info && !ssl_info->uriSanPeerCertificate().empty()) {
+      peer_san = ssl_info->uriSanPeerCertificate()[0];
+    }
+    absl::string_view peer_namespace;
+    if (!peer_san.empty()) {
+      const auto san_namespace = Utils::GetNamespace(peer_san);
+      if (san_namespace) {
+        peer_namespace = san_namespace.value();
+      }
+    }
+    if (peer_namespace.empty() && peer) {
```

Comment:
In which scenarios, will the variable peer be nil? 
If peer_namespace=="foo" and peer == nil, peer_namespace will be unset. 
Maybe checking peer != nil in a separate checking statement that is immediately after peer is set?

Reply:
Peer is nil when there is no metadata exchange (either header, or TCP prefix), and it's set on L732 and used extensively below. It's OK to have peer_namespace empty if there is no MX and no SAN namespace, it's handled with `!peer_namespace.empty()`.